R/catmaid_connection.R

```diff
@@ -534,3 +538,23 @@ set_catmaid_version <- function(version, conn=NULL, ...) {
   conn$catmaid.version=version
   catmaid_cache_connection(conn)
 }
+
+#This local function returns the string to be used in matching the environmental variable..
+# either catmaid_ or catmaid.
+catmaid_envstr <- function(){
+  tempstr <- paste0("^", "catmaid", "(\\.|_)") #searching for strings beginning with catmaid_ or catmaid.
+  matchvalues <- grep(pattern = tempstr, x = names(Sys.getenv()), value = TRUE)
+  
+  if (length(grep(pattern = paste0("^", "catmaid", "\\.") , x= matchvalues)) == 4) {
+    return(paste0("catmaid."))
+  } else if (length(grep(pattern = paste0("^", "catmaid", "_") , x= matchvalues)) == 4) {
+    return(paste0("catmaid_"))
+  }else if (length(grep(pattern = paste0("^", "catmaid", "\\.") , x= matchvalues)) == 0 && 
+            length(grep(pattern = paste0("^", "catmaid", "_") , x= matchvalues)) == 0) {
+    stop(paste("\ncatmaid error: No usable environmental variables found"))
+  } 
```

Comment:
Don't error out here – if you look at travis you'll see that this code fails in the absence of appopriately defined environment variables.

Reply:
Also 
```
stop("catmaid error: No usable environmental variables found")
```
should be ok. The paste isn't doing anything. Also why the leading `\n`?

Reply:
I have updated this as a simplemessage in line 552 in the latest commit and I have also added testcases for the same in the tests.